include/igraph_datatype.h

```diff
@@ -30,6 +30,43 @@
 
 __BEGIN_DECLS
 
+struct igraph_i_property_cache_t;
+typedef struct igraph_i_property_cache_t igraph_i_property_cache_t;
+
+typedef enum {
+    /* Stores whether the graph has at least one self-loop. */
+    IGRAPH_PROP_HAS_LOOP = 0,
+
+    /* Stores whether the graph has at least one multi-edge, taking into account
+     * edge directions in directed graphs. In other words, this property should
+     * be false for a directed graph with edges (a, b) and (b, a), and true
+     * for a directed graph with edges (a, b) and (a, b) again. */
+    IGRAPH_PROP_HAS_MULTI,
+
+    /* Stores whether the graph has at least one reciprocal edge pair. Ignored
+     * in undirected graphs. This property should be true for a directed graph
+     * with edges (a, b) and (b, a), and false for a directed graph with
+     * edges (a, b) and (a, b) again. */
```

Comment:
```suggestion
    /* Stores whether the graph has at least one reciprocal edge pair. Ignored
     * in undirected graphs. This property should be true for a directed graph
     * with edges (a, b) and (b, a), and false for a directed graph with
     * edges (a, b) and (a, b) again. Self-loops (a, a) are not considered
     * reciprocal. */
```

This is so as to keep this useful for determining whether a directed graph would have multi-edges when ignoring edge directions.